### PR TITLE
Hooks mounted as Inputs

### PR DESCRIPTION
**Pull request by swistach**
### Context
I describe the problem in #83.

### How has this been tested?
Define hook as an arrow function:
```
beforeChanges: (hotInstance, changes, source) => {
  // this -> should be your context (probably your component)

  return false; // This line should block inserting of changes.
};
```

### Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Related issue(s):
1. #83

### CODE REVIEW

File: lib/pro/src/hot-settings-resolver.service.ts [hunk 1]
```diff
@@ -1,120 +1,34 @@
...
-'beforeOnCellMouseOver', 'beforePaste', 'beforeRedo', 'beforeRemoveCellMeta', 'beforeRemoveCol',
-'beforeRemoveRow', 'beforeRender', 'beforeRenderer', 'beforeRowMove', 'beforeRowResize', 'beforeSetRangeEnd',
-'beforeSetRangeEnd', 'beforeStretchingColumnWidth', 'beforeTouchScroll', 'beforeUndo', 'beforeValidate',
-'beforeValueRender', 'construct', 'init', 'manualRowHeights', 'modifyAutofillRange', 'modifyCol',
-'modifyColHeader', 'modifyColumnHeaderHeight', 'modifyColWidth', 'modifyCopyableRange', 'modifyData',
-'modifyRow', 'modifyRowData', 'modifyRowHeader', 'modifyRowHeaderWidth', 'modifyRowHeight', 'modifyTransformEnd',
-'modifyTransformStart', 'persistentStateLoad', 'persistentStateReset', 'persistentStateSave',
-'skipLengthCache', 'unmodifyCol', 'unmodifyRow'];
+const AVAILABLE_OPTIONS: string[] = Object.keys(Handsontable.DefaultSettings.prototype);
+const AVAILABLE_HOOKS: string[] = Object.keys(Handsontable.hooks.globalBucket);
```
Comment (budnix) [inline lib/pro/src/hot-settings-resolver.service.ts]: Did you try `Handsontable.hooks.getRegistered()` method? It returns all you need in this case an array of strings of all available hooks.

File: lib/ce/src/hot-settings-resolver.service.ts [hunk 2]
```diff
@@ -47,44 +47,30 @@ const AVAILABLE_HOOKS: string[] = ['afterBeginEditing', 'afterCellMetaReset', 'a
...
-    if (component['settings'] !== void 0) {
-      Object.keys(component['settings']).forEach((key) => {
-        if (this.hooks.indexOf(key) > -1) {
-          mergedSettings[key] = (p1, p2, p3, p4, p5, p6) => {
-            return component._ngZone.run(() => {
-              return component['settings'][key].call(component.hotInstance, p1, p2, p3, p4, p5, p6);
-            })
-          };
+    options.forEach((key) => {
+      let option = component['settings'] && component['settings'][key];
```
Comment (budnix) [inline lib/ce/src/hot-settings-resolver.service.ts]: This expression can be risky in cases where `component['settings']` contains different value than an object. Could you secure this by adding `typeof` or something similar to the equalization?

File: lib/ce/src/hot-settings-resolver.service.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { Injectable, SimpleChanges } from '@angular/core';
 
-const AVAILABLE_OPTIONS: string[] = ['allowEmpty', 'allowHtml', 'allowInsertColumn', 'allowInsertRow',
+const AVAILABLE_OPTIONS: string[] = ['activeHeaderClassName', 'allowEmpty', 'allowHtml', 'allowInsertColumn', 'allowInsertRow',
```
Comment (budnix) [inline lib/ce/src/hot-settings-resolver.service.ts]: This shouldn't be replaced with 
```
const AVAILABLE_OPTIONS: string[] = Object.keys(Handsontable.DefaultSettings.prototype);
const AVAILABLE_HOOKS: string[] = Object.keys(Handsontable.hooks.globalBucket);
```
as in pro version?

Comment (swistach) [reply lib/ce/src/hot-settings-resolver.service.ts]: We have to fix our definition file first.

**Pull request merged**